app/assets/javascripts/discourse/app/lib/sidebar/common/community-section/everything-section-link.js

```diff
@@ -49,16 +48,23 @@ export default class EverythingSectionLink extends BaseSectionLink {
   }
 
   get currentWhen() {
-    if (this.linkToNew) {
-      return "discovery.new";
+    if (this.#linkToNew) {
+      if (this.#unreadAndNewCount > 0) {
+        return "discovery.new";
+      } else {
+        return "discovery.latest";
+      }
```

Comment:
I have the same concerns as https://github.com/discourse/discourse/pull/20488/files#r1122368918 here